examples/small-study/src/background.js

```diff
@@ -0,0 +1,76 @@
+// Allow outside override for debug and testing using prefs
+const STUDYPREFS = {
+  variationPref: `shield.${addonWidgetId}.testingBranch`,
+  firstrunPref: `shield.${addonWidgetId}.firstrun`,
+};
+
+const STUDYCONFIG = {
+  // the study
+  weightedVariations: {},
+};
+
+// This is a Shield Study template.
+class BaseStudy {
+  constructor(variation) {
+    // also sets activeExperiment key
+    // makes telmemetry possible
+    browser.study.configure(variation, ...STUDYCONFIG);
+  }
+
+  async isEligible() {
+    let permissions = await browser.study.permissions();
+    // could have other reasons to be eligible, such as addons or whatever
+    return permissions.shield;
+  }
+
+  // TODO gets reasons
+  // https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/runtime/onInstalled
+  async installOrDie() {
+    let eligible = await this.isEligible();
+    if (!eligible) {
+      // a full ineligible ending TODO
+      browser.study.endStudy("ineligible" /*, ...*/);
+    }
+  }
+
+  async watchExpire() {
+    // TODO, use the firstRun prefs, timers module etc.
+    let becameExpired = false;
+    if (becameExpired) {
+      // full ending
+      await browser.study.endStudy("expired" /*, ...*/);
+    }
+  }
+
+  watchImportantPrefs() {
```

Comment:
I like this. It does however require something like browser.study.watchPreferences() or similar to the API which is currently missing

Reply:
This also seems like it should be in the studyUtils API (at least the preferences we end the study for that are common to all studies like auto private browsing mode).

Reply:
I think a better solution here is "watchPermissions".   Prefs.watch belongs over in prefs.